core/src/main/java/org/apache/calcite/plan/RelOptPredicateList.java

```diff
@@ -245,4 +248,36 @@ public boolean isEffectivelyNotNull(RexNode e) {
     }
     return false;
   }
+
+  /** Return the set of columns that are set to a constant or scalar. */
+  public ImmutableBitSet getInvariantColumnSet() {
```

Comment:
I am not committed to the word 'invariant'. I was looking for a word that describes something known to be constant that isn't necessarily a constant literal.